webui/react/src/stores/experiments.tsx

```diff
@@ -1,63 +1,99 @@
-  if (context === null) {
-    throw new Error('Attempted to use useFetchExperiments outside of Experiment Context');
+  private constructor() {
+    if (!ExperimentsService.#isInternalConstructing) {
+      throw new TypeError('ExperimentsService is not constructable');
+    }
   }
-  const { experimentsCache, updateExperimentsCache } = context;
-
-  return useCallback(async (): Promise<void> => {
-    try {
-      const response = await getExperiments(params, { signal: canceler.signal });
-      updateExperimentsCache(experimentsCache.set(encodeParams(params), response));
-    } catch (e) {
-      handleError(e);
+
+  public static getInstance(): ExperimentsService {
+    if (!ExperimentsService.#instance) {
+      ExperimentsService.#isInternalConstructing = true;
+      ExperimentsService.#instance = new ExperimentsService();
+      ExperimentsService.#isInternalConstructing = false;
     }
-  }, [canceler, experimentsCache, params, updateExperimentsCache]);
-};
+    return ExperimentsService.#instance;
+  }
 
-export const useExperiments = (params: GetExperimentsParams): Loadable<ExperimentPagination> => {
-  const context = useContext(ExperimentsContext);
-  if (context === null) {
-    throw new Error('Attempted to use useExperiments outside of Experiment Context');
+  public experimentsByParams(
+    params: Readonly<GetExperimentsParams>,
+  ): Observable<Loadable<ExperimentPagination>> {
+    return this.#experimentsCache.select((map) => {
+      const cache = map.get(encodeParams(params));
+      if (!cache) {
+        return NotLoaded;
+      }
+      const experiments: ExperimentItem[] = [];
+      const expMap = this.#experimentMap.get();
+      for (const id of cache.experimentIds) {
+        const expItem = expMap.get(id);
+        if (expItem) {
+          experiments.push(expItem);
+        }
+      }
+      const expPagination: ExperimentPagination = {
+        experiments: experiments,
+        pagination: cache.pagination,
+      };
+      return Loaded(expPagination);
+    });
   }
-  const loadedVal = context.experimentsCache.get(encodeParams(params));
 
-  return loadedVal ? Loaded(loadedVal) : NotLoaded;
-};
+  public fetchExperiments(
+    params: Readonly<GetExperimentsParams>,
+    canceler: AbortController,
+  ): () => Promise<void> {
+    return async () => {
+      try {
+        const response = await getExperiments(params, { signal: canceler.signal });
+        this.#updateexperimentsCache(response, params);
+        this.#updateExperimentMap(response);
+      } catch (e) {
+        handleError(e);
+      }
+    };
+  }
+
+  #updateExperimentMap(expPagination: Readonly<ExperimentPagination>) {
+    this.#experimentMap.update((map) => {
+      let newMap: Map<number, ExperimentItem> = Map<number, ExperimentItem>();
+      for (const exp of expPagination.experiments) {
+        newMap = map.set(exp.id, exp);
+      }
+      return newMap;
+    });
+  }
+
+  #updateexperimentsCache(
```

Comment:
super nit: capitalize the `e` in experiments.

Also, the same class private methods problem as above.